lib/logidze/versioned_association.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+require 'active_support'
+
+module Logidze
+  module VersionedAssociation
+    module SingularAssociation
+      def reader(force_reload = false)
```

Comment:
@palkan I've got an idea. We override only `load_path` and `association`, as it was before my last commit, but in `association` we add something like this:

```
if association.is_a?(ActiveRecord::Associations::SingularAssociation) &&
   association.loaded? &&
   !association.inversed
  # we can extract functionality from load_target and use it here
  association.target.at! @logidze_requested_ts
end
```

What do you think about it? The same level of monkey-patching and no reload/reset confusion :)